pkg/operator/controllers/checker/unsupportconditionchecker.go

```diff
@@ -0,0 +1,122 @@
+package checker
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	configv1 "github.com/openshift/api/config/v1"
+	consolev1 "github.com/openshift/api/console/v1"
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	consoleclient "github.com/openshift/client-go/console/clientset/versioned"
+	"github.com/openshift/library-go/pkg/config/clusteroperator/v1helpers"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	clusterOperatorName = "aro"
+	reasonAsExpected    = "AsExpected"
+)
+
+type NodesChecker struct {
+	kubernetescli kubernetes.Interface
+	consolecli    consoleclient.Interface
+	configcli     configclient.Interface
+}
+
+func NewNodesChecker(kubernetescli kubernetes.Interface, consolecli consoleclient.Interface, configcli configclient.Interface) *NodesChecker {
+	return &NodesChecker{
+		kubernetescli: kubernetescli,
+		consolecli:    consolecli,
+		configcli:     configcli,
+	}
+}
+
+func (nc *NodesChecker) Name() string {
+	return "WorkerNodesCountChecker"
+}
+
+func (nc *NodesChecker) Check(ctx context.Context) error {
+	bannerName := "aro-sre"
+	nodes, err := nc.kubernetescli.CoreV1().Nodes().List(ctx, metav1.ListOptions{
+		LabelSelector: "node-role.kubernetes.io/worker",
+	})
+	if err != nil {
+		return err
+	}
+
+	if len(nodes.Items) < 3 {
+		// Set the Cluster Operator Upgradable status to "False"
+		nc.setCluOperatorStatus(ctx, 0)
+
+		// Call the newBanner function to display notification on the banner
+		banner := nc.newBanner(bannerName, "The number of the current worker nodes are less then 3")
+
+		_, err := nc.consolecli.ConsoleV1().ConsoleNotifications().Get(ctx, bannerName, metav1.GetOptions{})
+		if err != nil && !kerrors.IsNotFound(err) {
+			return err
+		}
+
+		// if the object doesn't exist Create
+		if err != nil && kerrors.IsNotFound(err) {
+			_, err = nc.consolecli.ConsoleV1().ConsoleNotifications().Create(ctx, banner, metav1.CreateOptions{})
+			return err
+		}
+
+		// if there's no errors, object found then update
+		_, err = nc.consolecli.ConsoleV1().ConsoleNotifications().Update(ctx, banner, metav1.UpdateOptions{})
+		return err
```

Comment:
This block should be added to the `newBanner` function, which should be refactored to be an `addConsoleBanner` function that appends the message to any existing messages.